packages/common/core-interfaces/src/brandedType.ts

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Base branded type which can be used to annotate other type.
+ *
+ * @remarks
+ * To use derive another class declaration and ideally add additional private
+ * properties to further distinguish the type.
+ *
+ * Since branded types are not real value types, they will always need to be
+ * created using `as` syntax and very often `as unknown` first.
+ *
+ * This class should never exist at runtime, so it is only declared.
```

Comment:
Nit:
```suggestion
 * @privateRemarks This class should never exist at runtime, so it is only declared.
```

Reply:
Fair. I guess info doesn't seem necessary given how it's used. But it's not harmful either.

What would really be useful is an `@example` block that _demonstrates_ how to use the type.